### PR TITLE
Update webdataset

### PR DESCRIPTION
**Pull request by nithinraok**
# What does this PR do ?

* Update version to 0.2.86
* Pipe functionality removed and has been replaced with DataPipeline. 

~~Add global_seed functionality to exp_manager for reproducibility.~~ <- will move to a separate PR 

**Collection**: ASR, TTS, NLP

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

### CODE REVIEW

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by titu1994** [comment]
jenkins

**Comment by titu1994** [comment]
jenkins

**Comment by pablo-garay** [comment]
jenkins

**Comment by pablo-garay** [comment]
jenkins

**Comment by pablo-garay** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by ericharper** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

File: Jenkinsfile [hunk 1]
```diff
@@ -392,6 +392,67 @@ pipeline {
 
     // TODO: this requires TE >= v0.11 which is not available in 23.06.
     //        please uncomment this test once mcore CI is ready.
+    
+    stage('Punctuation & Capitalization tarred dataset') {
```
Comment (nithinraok) [inline Jenkinsfile:396]: Before merging, I will replace this stages position to it original location.

**Comment by ericharper** [comment]
jenkins

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

File: Jenkinsfile (hunk 1 above)
Comment (titu1994) [reply Jenkinsfile:403]: ^^

File: nemo/collections/asr/data/audio_to_text.py [hunk 2]
```diff
@@ -994,6 +997,7 @@ def _compute_len(self):
             logging.info(f'Sharded manifests: Total length: {my_len}')
         else:
             my_len = len(self.manifest_processor.collection)
+            print("len: ", my_len)
```
Comment (titu1994) [inline nemo/collections/asr/data/audio_to_text.py:1000]: Remove prints

File: nemo/utils/exp_manager.py [hunk 3]
```diff
@@ -29,6 +29,7 @@
 from hydra.core.hydra_config import HydraConfig
 from hydra.utils import get_original_cwd
 from omegaconf import DictConfig, OmegaConf, open_dict
+from pytorch_lightning import seed_everything
```
[the remark this answers is not in the document]
Comment (titu1994) [reply nemo/utils/exp_manager.py:32]: Import not needed, lets do it in another pr

**Review by titu1994 — COMMENTED** [review]
Looks very good ! Just minor cleanup left

**Review by titu1994 — DISMISSED** [review]
Thanks!

**Pull request merged**